core-api/src/main/java/org/silverpeas/core/backgroundprocess/BackgroundProcessTask.java

```diff
@@ -63,37 +80,19 @@
 
   @SuppressWarnings("unchecked")
   private static void updateContextWith(final AbstractBackgroundProcessRequest request) {
-    final boolean hasToPush;
     synchronized (synchronizedContexts) {
       RequestContext context = synchronizedContexts.get(request.getUniqueId());
-      if (context == null) {
+      if (context == null || context.isRemovable()) {
         final RequestContext newRequestContext = new RequestContext(request);
         synchronizedContexts.put(request.getUniqueId(), newRequestContext);
-        hasToPush = true;
-        getLogger()
-            .debug(() -> format("pushing a not yet referenced process {0}", newRequestContext));
+        getLogger().debug(() -> format("pushing a new process {0}", newRequestContext));
+        RequestTaskManager.push(BackgroundProcessTask.class, request);
       } else {
-        if (!context.hasBeenProcessed()) {
-          // Request to process is updated
-          context.setRealRequest(request);
-          hasToPush = false;
-          getLogger().debug(() -> format("replacing a referenced process {0}", context));
-        } else if (context.isRemovable()) {
-          final RequestContext newRequestContext = new RequestContext(request);
-          synchronizedContexts.put(request.getUniqueId(), newRequestContext);
-          hasToPush = true;
-          getLogger().debug(() -> format("replacing referenced process, but no more valid, {0}",
-              newRequestContext));
-        } else {
-          getLogger().debug(() -> format(
-              "ignoring a process because it is yet valid and it has been processed {0}", context));
-          hasToPush = false;
-        }
+        getLogger().debug(() -> format(
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Format specifiers should be used instead of string concatenation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS3457)